tick/stateful/eval_function_node.go

```diff
@@ -32,15 +33,35 @@ func NewEvalFunctionNode(funcNode *ast.FunctionNode) (*EvalFunctionNode, error)
 }
 
 func (n *EvalFunctionNode) Type(scope ReadOnlyScope, executionState ExecutionState) (ast.ValueType, error) {
-	// PERF: today we are evaluating the function, it will be much faster if will type info the function it self
-	result, err := n.callFunction(scope.(*Scope), executionState)
-	if err != nil {
-		return ast.InvalidType, err
+	f := executionState.Funcs[n.funcName]
+	if f == nil {
+		df := scope.DynamicFunc(n.funcName)
+		if df == nil {
+			return ast.InvalidType, fmt.Errorf("undefined function: %q", n.funcName)
+		}
+		f = df
+	}
+	signature := f.Signature()
+
+	domain := Domain{}
+	if len(n.argsEvaluators) > len(domain) {
+	}
```

Comment:
TBH, no idea. I think I meant for there to be an error if someone supplied more arguments than any of our functions took, but it looks like I never got around to filling that in. I'll do that now.